### PR TITLE
build: drop -Wno-missing-field-initializers

### PR DESCRIPTION
**Pull request by t-8ch**

### CODE REVIEW

File: disk-utils/cfdisk.c [hunk 1]
```diff
@@ -195,17 +195,17 @@ struct cfdisk_menu {
...
-	{ 'n', N_("New"), N_("Create new partition from free space") },
-	{ 'q', N_("Quit"), N_("Quit program without writing changes") },
-	{ 't', N_("Type"), N_("Change the partition type") },
-	{ 'h', N_("Help"), N_("Print help screen") },
-	{ 's', N_("Sort"), N_("Fix partitions order") },
-	{ 'W', N_("Write"), N_("Write partition table to disk (this might destroy data)") },
-	{ 'u', N_("Dump"), N_("Dump partition table to sfdisk compatible script file") },
-	{ 0, NULL, NULL }
+	{ 'b', N_("Bootable"), N_("Toggle bootable flag of the current partition"), NULL },
+	{ 'd', N_("Delete"), N_("Delete the current partition"), NULL },
```
Comment (karelzak) [inline disk-utils/cfdisk.c:199]: Why do we need NULL here? I guess all unspecified fields are initialized to NULL/0.

File: include/pt-mbr-partnames.h [hunk 2]
```diff
@@ -2,116 +2,116 @@
...
-	{0x65, N_("Novell Netware 386")},
-	{0x70, N_("DiskSecure Multi-Boot")},
-	{0x75, N_("PC/IX")},
-	{0x80, N_("Old Minix")},	/* Minix 1.4a and earlier */
-	{0x81, N_("Minix / old Linux")},/* Minix 1.4b and later */
-	{0x82, N_("Linux swap / Solaris")},
-	{0x83, N_("Linux")},
-	{0x84, N_("OS/2 hidden or Intel hibernation")},/* OS/2 hidden C: drive,
+	{.code = 0x00, N_("Empty")},
+	{.code = 0x01, N_("FAT12")},
```
Comment (karelzak) [inline include/pt-mbr-partnames.h:6]: That's strange to mix named and unnamed initialization.

File: libmount/python/context.c [hunk 3]
```diff
@@ -1175,43 +1175,17 @@ static PyObject *Context_repr(ContextObjext *self)
...
+	.tp_dealloc = (destructor)Context_dealloc,
+	.tp_repr = (reprfunc) Context_repr,
+	.tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
+	.tp_doc = Context_HELP,
+	.tp_methods = Context_methods,
+	.tp_members = Context_members,
+	.tp_getset = Context_getseters,
+	.tp_init = (initproc)Context_init,
+	.tp_new = Context_new,
 };
```
Comment (karelzak) [inline libmount/python/context.c:1189]: This is a good idea; initializing such large structures without member names is ugly.

File: libmount/src/optmap.c [hunk 4]
```diff
@@ -70,56 +70,56 @@
  */
 static const struct libmnt_optmap linux_flags_map[] =
 {
-   { "ro",       MS_RDONLY },                 /* read-only */
+   { "ro",       MS_RDONLY, 0 },              /* read-only */
```
Comment (karelzak) [inline libmount/src/optmap.c:73]: Again, why? C standard is pretty obvious about unspecified field initialization.

**Comment by karelzak** [comment]
I'm not sure about this PR.

Not sure about the latest gcc, but  -Wno-missing-field-initializers was a good idea for GCC :-)

**Comment by t-8ch** [comment]
This PR does not fix real, current defects.
The goal was more to get rid of the disablement of the warning, so potential new issues are shown.

If you dislike it I can open a new PR with just the uncontroversial parts.

**Comment by t-8ch** [comment]
Closing as not wanted. Acceptable parts have already been merged.

**Pull request closed without merging**